training_main.py

```diff
@@ -0,0 +1,354 @@
+    if embedding_vector is not None:
+        embedding_matrix[i]=embedding_vector
+        
+print("Embedding Matrix dimensions : ", embedding_matrix.shape)
+
+# %%
+
+# Customizing the layers of the model and selecting the appropriate activation functions
+
+inputs1=Input(shape=(2048,))
+fe1=Dropout(0.5)(inputs1)
+fe2=Dense(256,activation='relu')(fe1)
+inputs2=Input(shape=(max_length,))
+
+se1=Embedding(vocab_size,embedding_dim,mask_zero=True)(inputs2)
+se2=Dropout(0.5)(se1)
+se3=LSTM(256)(se2)
+decoder1=add([fe2,se3])
+decoder2=Dense(256,activation='relu')(decoder1)
+outputs=Dense(vocab_size,activation='softmax')(decoder2)
+model=Model(inputs=[inputs1,inputs2],outputs=outputs)
+
+# %%
+
+# Printing the summary of the model
+
+model.summary()
+
+# %%
+
+# Setting the weights of the model layer in accordance with the 'embedding-matrix' common wordset
+
+model.layers[2]
+model.layers[2].set_weights([embedding_matrix])
+model.layers[2].trainable=False
+
+# %%
+
+# Compiling the model (selecting loss function and 'adam' optimizer)
+
+model.compile(loss='categorical_crossentropy',optimizer='adam')
+
+# %%
+
+# Setting parameters for model training/optimization
+
+epochs=9
+number_pics_per_batch=3
+steps=len(descriptions)//number_pics_per_batch
+
+# %%
+
+del descriptions['image_name']
+del descriptions['']
+
+# %%
+
+# Saving the model(first instance) locally
+model.save('./model_weights/model_'+str(0)+'.h5')
+
+# %%
+
+# Optimizing the model weighs (epoch number of times) and saving the weights locally each time
+
+for i in range(epochs+1):
+    generator=training_functions.data_generator(descriptions,train_features,wordtoix,max_length,number_pics_per_batch)
+    model.fit_generator(generator,epochs=1,steps_per_epoch=steps,verbose=1)
+    model.save('./model_weights/model_'+str(i)+'.h5')
+
+# %%
+
+# Storing the final trained model in 'new_model'
+    
+import tensorflow as tf
+new_model=tf.keras.models.load_model('./model_weights/model_7.h5')
+
+# %%
+
+# Setting new parameters for model optimization
+
+model.compile(loss='categorical_crossentropy',optimizer=tf.compat.v1.train.AdamOptimizer(learning_rate=0.0001))
+epochs = 10
+number_pics_per_bath = 6
+steps = len(descriptions)//number_pics_per_bath
+
+# %%
+
+# # Optimizing the model weighs (epoch number of times) and saving the weights locally each time
+
+for i in range(epochs+1):
+    generator=training_functions.data_generator(descriptions,train_features,wordtoix,max_length,number_pics_per_batch)
+    model.fit_generator(generator,epochs=1,steps_per_epoch=steps,verbose=1)
+    model.save('./model_weights/model_'+str(i+10)+'.h5')
+
+# %%
+
+# Initializing final model with final-trained weights
+    
+import tensorflow as tf
+model=tf.keras.models.load_model('./model_weights/final_model.h5')
+model.compile(loss='categorical_crossentropy',optimizer=tf.compat.v1.train.AdamOptimizer(learning_rate=0.0001))
+
+# %%
+
+# Initializing the path of the test images
+
```

Comment:
this is the part where predictions are being made which is the part of the predict file and it should not be here. remove the prediction part from here.